driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/ChangeStreamPublisher.java

```diff
@@ -135,6 +136,26 @@ public interface ChangeStreamPublisher<TResult> extends Publisher<ChangeStreamDo
      */
     ChangeStreamPublisher<TResult> batchSize(int batchSize);
 
+    /**
+     * Sets the timeoutMode for the cursor.
+     *
+     * <p>
+     *     Requires the {@code timeout} to be set, either in the {@link com.mongodb.MongoClientSettings},
+     *     via {@link MongoDatabase} or via {@link MongoCollection}
+     * </p>
+     * <p>
+     *     If the {@code timeout} is set then:
+     *     <ul>
+     *      <li>For non-tailable cursors, the default value of timeoutMode is {@link TimeoutMode#CURSOR_LIFETIME}</li>
+     *      <li>For tailable cursors, the default value of timeoutMode is {@link TimeoutMode#ITERATION} and its an error
+     *      to configure it as: {@link TimeoutMode#CURSOR_LIFETIME}</li>
+     *     </ul>
+     * @param timeoutMode the timeout mode
+     * @return this
+     * @since 4.x
+     */
+    ChangeStreamPublisher<TResult> timeoutMode(TimeoutMode timeoutMode);
```

Comment:
As I have noticed in the spec, change streams should not support timeout mode as they always use `ITERATION`.

> These helpers MUST NOT support the timeoutMode option as change streams are an abstraction around tailable-awaitData cursors, so they implicitly use ITERATION mode

It looks safe to remove this method from this specific interface. However, in the case of the synchronous version, `ChangeStreamIterable` extends `MongoIterable`, which includes the definition of `timeoutMode`, so `ChangeStreamIterableImpl` ends up inheriting it.

Considering timeout mode isn't relevant for change streams, what do you think about not defining the `timeoutMode()` method in `MongoIterable` and instead placing it on `Iterables` where it can be used? It might avoid a bit of confusion for users since having this method on change streams, where it can't be utilized, might be a tad confusing.

Reply:
Done - only tricky part was adding some reflection in the UnifiedCrudHelper